How-To-Data-Query-Templates.md

```diff
@@ -0,0 +1,200 @@
+# Data Query Templates
+
+During the past few weeks as I've made various SNMP Data Query templates, I've
+ran into a few 'tricks' (which existing user scripts use) to make them work
+with most devices. It'd probably be beneficial to publish these in some manner
+on the documentation site so others don't have to stumble.
+
+As you already know, getting an index is the hardest thing users struggle with.
+I've ran across three scenarios and their solutions:
+
+## SNMP index is present
+
+Basically mimic the interface.xml template which comes with Cacti
+`(cacti\resource\snmp_queries\interface.xml)`.
+
+SNMPWALK:
+
+```console
+Raw OIDs:
+.1.3.6.1.2.1.2.2.1.1.1 = INTEGER: 1
+.1.3.6.1.2.1.2.2.1.1.2 = INTEGER: 2
+.1.3.6.1.2.1.2.1.0 = INTEGER: 2
+```
+
+```console
+Textual MIB names:
+IF-MIB::ifIndex.1 = INTEGER: 1
+IF-MIB::ifIndex.2 = INTEGER: 2
+IF-MIB::ifNumber.0 = INTEGER: 2
+```
+
+In the SNMP XML file, specify the `<oid_index>` and optional
+`<oid_num_indexes>` fields. For example:
+
+```XML
+<interface>
+    <name>Get SNMP Interfaces</name>
+    <description>Queries a host for a list of monitorable interfaces</description>
+    <oid_index>.1.3.6.1.2.1.2.2.1.1</oid_index>
+    <oid_num_indexes>.1.3.6.1.2.1.2.1.0</oid_num_indexes>
+
+    <fields>
+        <ifIndex>
+            <name>Index</name>
+            <method>walk</method>
+            <source>value</source>
+            <direction>input</direction>
+            <oid>.1.3.6.1.2.1.2.2.1.1</oid>
+        </ifIndex>
+    </fields>
+</interface>
+```
+
+A verbose query from inside Cacti shows the following:
+
+```markdown
```

Comment:
should be shell not markdown as it's console output

Reply:
in web browser, but still debug console output

Reply:
Disagree. That's cacti outputs from web gui not shell nor console but if you want it to be console. I'll make it console..

Reply:
Well, it's definitely not markdown 👍   and we don't allow the unrendered HTML.  If you think there is a different language selection that would work suggest away...  Here's where my logic comes from:  

The difference between console and shell is that console displays blue text, where shell is just your standard grey output (not sure what it highlights).  

Since it's just a fixed width output display with debug lines, I think shell works better since it isn't commands you are typing, but I'm not going to lose sleep over the difference (@ronytomen might though heh).